unified-runtime/source/adapters/level_zero/v2/command_buffer.hpp

```diff
@@ -51,11 +54,22 @@ struct ur_exp_command_buffer_handle_t_ : public ur_object {
       uint32_t numUpdateCommands,
       const ur_exp_command_buffer_update_kernel_launch_desc_t *updateCommands);
 
+  ur_exp_command_buffer_sync_point_t getSyncPoint(ur_event_handle_t event);
+  ur_event_handle_t *getWaitListFromSyncPoints(
+      const ur_exp_command_buffer_sync_point_t *pSyncPointWaitList,
+      uint32_t numSyncPointsInWaitList);
+
 private:
+  std::vector<ur_event_handle_t> syncPoints;
```

Comment:
Do we need to be using `ur_event_handle_t`s for this, or could we store L0 events directly? The UR user is going to be interacting with sync-point types, not ur events, so there's an opportunity to use the lower level type to avoid some overhead.

I'm not familiar enough with the v2 adapter to know if this is a pain to do with the manager interfaces. If so, then this is fine as-is, just a suggestion.

Reply:
I think it's easier to use `ur_event_handle_t` because of the added ref counting. These are faux-RAII and return automatically to the pool they were allocated from.